### PR TITLE
fix: sd-input a11y

### PR DESCRIPTION
**Pull request by smfonseca**
## Description:
This PR closes [1479](https://github.com/solid-design-system/solid/issues/1479) and [1520](https://github.com/solid-design-system/solid/issues/1520)
It also introduces custom spin buttons for number inputs which improves the component accessibility since now all changes in a type="number" input are correctly announced by screen readers.

@MartaPintoTeixeira @coraliefeil in this PR there are copy changes, mainly for `max-length` and `min-length`. Please consider them to align with Figma. If you have any questions, please let me know.
-> Checked. No todos on our end needed.

## Definition of Reviewable:

- [x] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [x] E2E tests (features, a11y, bug fixes) are created/updated

- [x] Stories (features, a11y) are created/updated
- [x] relevant tickets are linked

### CODE REVIEW

**Comment by coraliefeil** [comment]
Checked. No todos on our end needed.

**Review by MartaPintoTeixeira — APPROVED** [review]

File: packages/components/src/components/input/input.stories.ts [hunk 1]
```diff
@@ -143,6 +143,23 @@ export const Clearable = {
...
 
+/**
+ * Use the `left` or `right` slot to add system icons.
+ */
+
+export const Icon = {
+  render: () =>
+    html`<div class="flex flex-rows gap-12">
+      <sd-input label="Label" spellcheck>
+        <sd-icon label="landscape" library="global-resources" name="system/picture" slot="left"></sd-icon>
```
Comment (auroraVasconcelos) [inline packages/components/src/components/input/input.stories.ts:154]: library="global-resources" is deprecated, no library is needed to be referenced

Comment (smfonseca) [reply packages/components/src/components/input/input.stories.ts:154]: Thanks for the information @auroraVasconcelos. Was this documented somewhere that I may have missed it? I still see lots of places in the codebase with this.

**Review by MartaPintoTeixeira — APPROVED** [review]

**Review by Vahid1919 — APPROVED** [review]

File: packages/components/src/components/input/input.stories.ts (hunk 1 above)
Comment (auroraVasconcelos) [reply packages/components/src/components/input/input.stories.ts]: @smfonseca, AFAIK this is not in our documentation. Maybe @mariohamann or @Vahid1919 may have more information on this.
I'd like to correct myself; earlier, I said _deprecated_ but actually, _library="global-resources"_ is legacy.

**Pull request merged**